portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/aui/ScriptData.java

```diff
@@ -229,6 +230,20 @@ private PortletData _getPortletData(String portletId) {
 		return portletData;
 	}
 
+	private Set<Map.Entry<String, PortletData>> _getPortletDataSet(
+		String portletId) {
+
+		Set<Map.Entry<String, PortletData>> portletDataSet =
+			_portletDataMap.entrySet();
+
+		if (Validator.isNotNull(portletId)) {
+			portletDataSet.removeIf(
```

Comment:
@pavel-savinov this is still really weird. Why are we removing other portlets' PortletData? And doing it twice?! (you call _getPortletDataSet() 2 times with the same portletId).
ScriptData is a servlet level thing, it contains multiple portlets' output. Are we sure we want to just wipe out others' output?